cpp/tests/reshape/byte_cast_tests.cpp

```diff
@@ -26,6 +26,23 @@ using namespace cudf::test;
 class ByteCastTest : public cudf::test::BaseFixture {
 };
 
+// Previously, there was a bug that causes thrust to crash
+// This test is to make sure that bug does not pop up again
+// If this test fails, the program will crash
+TEST_F(ByteCastTest, thurstTest)
```

Comment:
Typo. But I actually don't think this is a good test -- it doesn't test any expectations. I don't think we should just test for crashes. Also, if `differences_message` is only used inside of other column utilities, and is not meant to be used externally, then I doubt it needs to be directly tested. 

```suggestion
TEST_F(ByteCastTest, thrustTest)
```

Reply:
This is also what I'm thinking about. Then I'm going to merge this test with an existing test, and add a comment line to clarify the added code. This way makes more sense to me.

Reply:
How can the crash be dependent on a test failure? That would imply that it is dependent on GoogleTest behavior.

Reply:
The crash point is in the function stringify_column_differences, which operates on the input data resulted from the comparison of two columns. Only when the two given columns are different then this function crashes.

All the existing tests only call CUDF_TEST_EXPECT_COLUMNS_EQUAL to test if two columns having the same data. When the two given columns have different data (test will fail), this triggers the function stringify_column_differences which eventually crashes.

Reply:
The appropriate black-box test here would be to use `CUDF_TEST_EXPECT_COLUMNS_EQUAL` on two columns that are _not_ equal which will then call `stringify_column_differences`. That way `stringify_column_differences` doesn't need to be in a public header.

Reply:
Also, this test shouldn't go in `byte_cast_tests` as it has nothing to do with byte casting.

Reply:
> The appropriate black-box test here would be to use `CUDF_TEST_EXPECT_COLUMNS_EQUAL` on two columns that are _not_ equal which will then call `stringify_column_differences`. You can use GTEST to capture stdout and inspect the contents. That way `stringify_column_differences` doesn't need to be in a public header.

But doing so we will have failed test. We always want to have no failed test, don't we?